src/Microsoft.Identity.Web/AuthorizeForScopesAttribute.cs

```diff
@@ -49,10 +49,7 @@ public override void OnException(ExceptionContext context)
         {
             if (context != null)
             {
-                MsalUiRequiredException? msalUiRequiredException =
-                    (context.Exception as MsalUiRequiredException)
-                    ?? (context.Exception?.InnerException as MsalUiRequiredException);
-
+                MsalUiRequiredException? msalUiRequiredException = FindMsalUiRequiredExceptionIfAny(context.Exception);
```

Comment:
Customers rethrow exceptions and pass Microsoft.Identity.Web exceptions in as inner exceptions.
In 0.2.1-preview, we've added the `MicrosoftIdentityWebChallengeUserException`, but we overlooked the scenario where it would be nested (for instance by the Graph SDK)